client/testutil/driver_compatible.go

```diff
@@ -24,6 +24,14 @@ func RequireConsul(t *testing.T) {
 	}
 }
 
+// RequireVault skips tests unless a Vault binary is available on $PATH.
+func RequireVault(t *testing.T) {
+	_, err := exec.Command("vault", "version").CombinedOutput()
```

Comment:
I recently learned about [os.LookPath](https://pkg.go.dev/os/exec#LookPath) which is also useful for capability detection.  No need to change, just sharing information.